app/src/main/cpp/skyline/input/npad_device.cpp

```diff
@@ -211,6 +238,26 @@ namespace skyline::input {
         nextEntry.status.raw = connectionState.raw;
     }
 
+    void NpadDevice::WriteNextEntry(NpadSixAxisInfo &info, NpadSixAxisState entry) {
+        auto &lastEntry{info.state.at(info.header.currentEntry)};
+
+        info.header.timestamp = util::GetTimeTicks();
```

Comment:
Do you wanna commonise this header code, we have it duplicated in three places so maybe an idea?

Reply:
This is used for all hid devices. So making it a class is definitively the right way to go